packages/generator/src/parser/edmx-to-vdm-common.ts

```diff
@@ -418,56 +419,49 @@ function parseReturnType(
   };
 }
 
-export function transformFunctionImports(
-  serviceMetadata: ParsedServiceMetadata,
-  entities: VdmEntity[],
-  complexTypes: VdmComplexType[],
+export function transformFunctionImportBase(
+  edmxFunctionImport: EdmxFunctionImportBase,
+  edmxParameters: EdmxParameter[],
+  swaggerDefinition: SwaggerPath | undefined,
   formatter: ServiceNameFormatter
-): VdmFunctionImport[] {
-  const edmxFunctionImports = serviceMetadata.edmx.functionImports;
-
-  return edmxFunctionImports.map(f => {
-    const functionName = formatter.originalToFunctionImportName(f.Name);
-    const functionImport = {
-      httpMethod: f['m:HttpMethod'].toLowerCase(),
-      originalName: f.Name,
-      functionName,
-      returnType: parseReturnType(f.ReturnType, entities, complexTypes),
-      parametersTypeName: toTypeNameFormat(`${functionName}Parameters`)
-    };
-
-    const swaggerDefinition = swaggerDefinitionForFunctionImport(
-      serviceMetadata,
-      functionImport.originalName,
-      functionImport.httpMethod
-    );
-
-    const parameters = f.Parameter.map(p => {
-      const swaggerParameter = swaggerDefinition
-        ? swaggerDefinition.parameters.find(param => param.name === p.Name)
-        : undefined;
-      return {
-        originalName: p.Name,
-        parameterName: formatter.originalToParameterName(f.Name, p.Name),
-        edmType: parseType(p.Type),
-        jsType: edmToTsType(p.Type),
-        nullable: isNullableParameter(p),
-        description: parameterDescription(p, swaggerParameter)
-      };
-    });
+): Omit<VdmFunctionImport, 'returnType' | 'httpMethod'> {
+  const functionName = formatter.originalToFunctionImportName(
+    edmxFunctionImport.Name
+  );
+  const functionImport = {
+    originalName: edmxFunctionImport.Name,
+    functionName,
+    parametersTypeName: toTypeNameFormat(`${functionName}Parameters`)
+  };
 
+  const parameters = edmxParameters.map(p => {
+    const swaggerParameter = swaggerDefinition
+      ? swaggerDefinition.parameters.find(param => param.name === p.Name)
+      : undefined;
     return {
-      ...functionImport,
-      parameters,
-      description: functionImportDescription(
-        swaggerDefinition,
-        functionImport.originalName
-      )
+      originalName: p.Name,
+      parameterName: formatter.originalToParameterName(
+        edmxFunctionImport.Name,
+        p.Name
+      ),
+      edmType: parseType(p.Type),
+      jsType: edmToTsType(p.Type),
```

Comment:
the attribute is called jsType but the method TsType? Which type is it?

Reply:
I also stumbled upon this. As it is a type I guess it is ts ¯\_(ツ)_/¯. I didn't want to change it, because it would be a breaking change again. I could add a tsType property and deprecate this one, not sure if it is worth it though. What do you think?

Reply:
Also, maybe this just doesn't belong in this PR?